packages/legend-query-builder/src/components/QueryBuilderResultPanel.tsx

```diff
@@ -128,7 +147,8 @@ const QueryBuilderGridResultContextMenu = observer(
         .filter(
           (n) =>
             (n as QueryBuilderPostFilterTreeConditionNodeData).condition
-              .columnState.columnName === projectionColumnState.columnName &&
+              .columnState.columnName ===
+              (projectionColumnName ?? projectionColumnState.columnName) &&
```

Comment:
why did we modify the check here?

Reply:
Necessary now since `getExistingPostFilterNode` is changed/updated calling `projectionColumnName: string | undefined` so it can filter multiple values properly, otherwise it will add different values to same projection column